### PR TITLE
Breaking: replace n-logger with Reliability Kit logger

### PR DESCRIPTION
**Pull request by rowanmanning**
This replaces n-logger usage with Reliability Kit logger. It's a breaking change because the module now logs to  rather than sending logs directly to Splunk. This should be a safe upgrade for most Customer Products apps.

Part of [CPREL-668](https://financialtimes.atlassian.net/browse/CPREL-668).

[CPREL-668]: https://financialtimes.atlassian.net/browse/CPREL-668?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: test/unit/lib/metrics/system/process.spec.js [hunk 1]
```diff
@@ -218,7 +218,7 @@ describe('lib/metrics/system/process', () => {
 				sinon.stub(global, 'setInterval').returns({
 					unref: sinon.stub()
 				});
-				sinon.stub(process, 'hrtime');
+				sinon.stub(process, 'hrtime').returns([0, 0]);
```
Comment (rowanmanning) [inline test/unit/lib/metrics/system/process.spec.js:221]: This was a weird rabbit hole, the tests weren't passing due to some Node.js internals. It seems like now we're calling `hrtime` more than expected and it throws an error if this stub doesn't have a default return value

**Review by joelcarr — APPROVED** [review]

File: test/unit/lib/metrics/system/process.spec.js (hunk 1 above)
Comment (joelcarr) [reply test/unit/lib/metrics/system/process.spec.js:221]: Good to know, thanks for sharing! Do you think it is worth looking into more?

**Review by CyntiBinti — APPROVED** [review]

File: test/unit/lib/metrics/system/process.spec.js (hunk 1 above)
Comment (rowanmanning) [reply test/unit/lib/metrics/system/process.spec.js:221]: This thing's so messed up, I'm struggling to work out what's happening. So some things I've learned after more investigation. I did all of this after reverting the change that this comment thread is on.

* Nothing in the `package-lock.json` diff between `main` and this branch leads me to think it's a dependency issue. There are a few minor things bumped but nothing that should impact this

* If you run `make test` the tests will fail, but if you run `make unit-test` they will pass. So it seems to be something to do with how `nyc` is instrumenting the code?

* If you run `make unit-test-coverage` the tests will fail, but if you add `--cache false` after `nyc` then they pass

* The tests also pass if you remove the nyc cache before running them: `rm -rf node_modules/.cache/nyc && make unit-test-coverage`

* Now that I've run the tests more times, they don't always fail in _exactly_ the same way. We get the following errors in a mix of different combinations:
   
    * The most common (E1), sometimes appears for multiple test cases:
      ```
       Uncaught TypeError: Cannot read properties of undefined (reading '0')
         at now (node:internal/perf/utils:15:13)
      ```

    * A weird one (E2):
      ```
       TypeError: Cannot read properties of undefined (reading '0')
         at Array.onTick (lib/metrics/system/process.js:3:239)
      ```
      `lib/metrics/system/process.js` line 3 is this:
      ```js
      const os = require('os');
      ```

    * A very rare but very weird (E3) one that indicates sometimes mocks aren't applying?
      ```
      TypeError: global.setInterval.restore is not a function
        at Context.<anonymous> (test/unit/lib/metrics/system/process.spec.js:228:24)
      ```

    * The most rare one (E4) only failed like this when I was testing more thoroughly below:
      ```
      Uncaught AssertionError: expected undefined to equal 1
        at Test.<anonymous> (test/unit/http.spec.old.js:87:109)
      ```

* I tried running the tests multiple times to see if there was a pattern to the errors that appear. There is not. Each of the series of test results below were run after an `rm -rf node_modules/.cache/nyc`:
  1. `PASS`, `E1`, `E1+E1`, `E4+E1+E1`, `E1+E1`, `E1+E1`, `E1+E1`
  2. `PASS`, `E1+E3`, `E1+E3`, `E1+E1`, `E1+E1`, `E1+E3`

* Upgrading `nyc` to `12.0.2` (the latest version in the 12.x release line) does not fix the issue

* Upgrading `nyc` to `15.1.0` **does fix the issue**. I ran the tests a lot of times, all fine.

I feel fairly confident after this further testing that the instrumentation that `nyc` 12.x provides is broken in some way. I'm really not sure why my (really quite unrelated) change triggered this? Perhaps there's a race condition in here somewhere, perhaps the new logger also requires `os` in one of the sub-dependencies?

So I feel like I'm happy to make this change. My only question to you reviewers is whether you think I should:

1. Keep the change in the tests and keep `nyc@12`
2. Ditch the change in the tests and upgrade to `nyc@15`

Comment (joelcarr) [reply test/unit/lib/metrics/system/process.spec.js:221]: 🧃Thanks so much for looking into this. This is a JUICY problem. Considering `nyc@12` is 5 years old we should probably bump to the latest version (which is only 3 years old 😆). This is a dependency that will likely be reconsidered in the future / along with this entire package.

**Review by joelcarr — APPROVED** [review]
image

**Pull request merged**